static/app/components/button.tsx

```diff
@@ -97,6 +97,10 @@ function BaseButton({
   const screenReaderLabel =
     ariaLabel || (typeof children === 'string' ? children : undefined);
 
+  const hasChildren = Array.isArray(children)
+    ? children.some(child => !!child)
+    : !!children;
+
```

Comment:
If `children` is undefined, then we remove the right margin on `<Icon />`:
<img width="58" alt="Screen Shot 2022-02-01 at 5 06 04 PM" src="https://user-images.githubusercontent.com/44172267/152077186-c161331c-6c0f-447f-a557-aad1047b9d5b.png">
<img width="58" alt="Screen Shot 2022-02-01 at 5 06 23 PM" src="https://user-images.githubusercontent.com/44172267/152077212-8c1d42bf-bff0-4608-b993-5b9fe0b3803d.png">

Currently we check for that with `!!children`. But this doesn't work when `children` is an array whose elements are all `undefined`/`nullish`/`falsy`, which is the case here ([see `<DropdownButtonV2 />`](https://github.com/getsentry/sentry/blob/vuluong/alert-actions-dropdown/static/app/components/dropdownButtonV2.tsx#L40-L57)).

Reply:
if the dropdown does not have children, what will be displayed in the dropdown menu when the user clicks on the button? will the button be disabled?

Reply:
hmmm that's a good point, added some code to do that: https://github.com/getsentry/sentry/pull/31537/commits/6d87ee7ffe6b02a99726e783631f6b5e75a9957d